array/DNAm/analysis/EWAS/summariseWithinCTDMPs.r

```diff
@@ -0,0 +1,272 @@
+##
+## ---------------------------------------------------------------------#
+
+#----------------------------------------------------------------------#
+# DEFINE PARAMETERS
+#----------------------------------------------------------------------#
+args <- commandArgs(trailingOnly = TRUE)
+dataDir <- args[1]
+refPath <- args[2]
+
+normData <- file.path(dataDir, "3_normalised/normalised.rdata")
+resPath <- file.path(dataDir, "4_analysis/EWAS")
+
+cellTypes <- c("Double-", "NeuN+", "Sox10+")
+
+#----------------------------------------------------------------------#
+# LOAD PACKAGES
+#----------------------------------------------------------------------#
+library(qqman)
+library(tidyr)
+library(ggplot2)
+
+#----------------------------------------------------------------------#
+# IMPORT DATA
+#----------------------------------------------------------------------#
+
+res<-list()
+for (each in cellTypes) {
+    load(file.path(resPath, paste0(each, "LM.rdata")))
+    res[[each]] <- outtab
+}
+rm(outtab)
+
+#----------------------------------------------------------------------#
+# REMOVE CROSS HYB & SNP PROBES
+#----------------------------------------------------------------------#
+
+## filter out cross hyb & SNP probes
+crosshyb <- read.table(file.path(refPath, "CrossHydridisingProbes_McCartney.txt"), stringsAsFactors = FALSE)
+tofilter <- read.csv(file.path(refPath, "EPICArrayProbesToFilter.csv"), stringsAsFactors = FALSE)
+snpProbes <- read.table(file.path(refPath, "SNPProbes_McCartney.txt"), stringsAsFactors = FALSE, header = TRUE)
+crosshyb2 <- read.csv(file.path(refPath, "Pidsley_SM1.csv"), stringsAsFactors = FALSE)
+snpProbes2 <- read.csv(file.path(refPath, "Pidsley_SM4.csv"), stringsAsFactors = FALSE)
+snpProbes3 <- read.csv(file.path(refPath, "Pidsley_SM5.csv"), stringsAsFactors = FALSE)
+snpProbes4 <- read.csv(file.path(refPath, "Pidsley_SM6.csv"), stringsAsFactors = FALSE)
+
+snpProbes <- snpProbes[which(snpProbes$DIST_FROM_MAPINFO < 10 & snpProbes$AF > 0.01), ]
+snpProbes2 <- snpProbes2[which(snpProbes2$AF > 0.01), ]
+snpProbes3 <- snpProbes3[which(snpProbes3$AF > 0.01), ]
+snpProbes4 <- snpProbes4[which(snpProbes4$AF > 0.01), ]
+
+dist <- cbind(abs(snpProbes4$VARIANT_END - snpProbes4$MAPINFO), abs(snpProbes4$VARIANT_START - snpProbes4$MAPINFO))
+dist <- apply(dist, 1, min)
+snpProbes4 <- snpProbes4[which(dist <= 10), ]
+
+remove <- unique(c(tofilter$IlmnID, crosshyb[, 1], snpProbes$IlmnID, snpProbes2$PROBE, snpProbes3$PROBE, snpProbes4$PROBE))
+remove <- intersect(remove, rownames(res[[1]]))
+
+if (length(remove) > 0) {
+    for (i in 1:3) {
+        res[[i]] <- res[[i]][-match(remove, rownames(res[[i]])), ]
+    }
+}
+
+
+# add gene annotation
+probeAnnot <- read.table(file.path(refPath, "EPIC.anno.GRCh38.tsv"), sep = "\t", header = TRUE, stringsAsFactors = FALSE)
+probeAnnot <- probeAnnot[match(rownames(res[[1]]), probeAnnot$probeID), ]
+probeAnnot$chrm <- gsub("chr", "", probeAnnot$chrm)
```

Comment:
Considering that this substitution is static (always searching for the static string "chr"), using a regular expression is slow. It's likely that you will get a speed up if you use the `fixed = TRUE` flag. Same goes for line 187. 

```R
probeAnnot[[chrm]] <- gsub("chr", "", probeAnnot[[chrm]], fixed = True)
```